### PR TITLE
Stricter linting

### PR DESCRIPTION
**Pull request by JosefWN**
An attempt at fixing https://github.com/shamblett/coap/issues/77, we can make the config more relaxed as part of this PR if we want, this is just a fairly strict suggestion/starting point for discussions.

There are a bunch of things the default/recommended lint configs don't catch, mostly because there is no consensus on whether to lint in way A or way B for a given issue, so the recommendations boil down to neither (where it could be better to just choose A or B for consistency).

Relaxed the validation a little in the config builder, for example it now requires correct input on bools (strings will not pass as bools even if they are "true" or "false") and the `dtlsBackend`, they will know if they get it wrong anyway at compile time?

### CODE REVIEW

**Review by JKRhb — APPROVED** [review]
Thank you, @JosefWN, these are great improvements :) See a couple of minor comments below.

File: lib/src/stack/coap_token_layer.dart [hunk 1]
```diff
@@ -9,9 +9,7 @@ import 'dart:math';
 
 import 'package:typed_data/typed_data.dart';
 
-import '../coap_config.dart';
-import '../coap_constants.dart';
-import '../coap_request.dart';
+import '../../coap.dart';
```
[the remark this answers is not in the document]
Comment (JosefWN) [reply lib/src/stack/coap_token_layer.dart]: Could be something caused by `dart fix --apply`, not intentional, fixed: https://github.com/shamblett/coap/pull/95/commits/ec979c4406db8839059f009f8327cae884b5788e

File: lib/src/stack/coap_reliability_layer.dart [hunk 2]
```diff
@@ -25,13 +25,17 @@ import 'coap_ilayer.dart';
...
+  ) {
     currentTimeout = _message!.ackTimeout;
   }
 
-  final DefaultCoapConfig? _config;
-  final CoapExchange _exchange;
-  final CoapMessage? _message;
+  late final DefaultCoapConfig? _config;
+  late final CoapExchange _exchange;
+  late final CoapMessage? _message;
```
[the remark this answers is not in the document]
Comment (JosefWN) [reply lib/src/stack/coap_reliability_layer.dart]: Ah that was sloppy of me, the linter is complaining about `_config` and `_message` not being set to `late`:

> Use late for private members with a non-nullable type.dart ([use_late_for_private_fields_and_variables](https://dart-lang.github.io/linter/lints/use_late_for_private_fields_and_variables.html))

... I think what the linter is cryptically suggesting is that the two variables are effectively not nullable, even though their type is (set once with a non-nullable value). I changed to non-nullable types instead: https://github.com/shamblett/coap/pull/95/commits/ec979c4406db8839059f009f8327cae884b5788e

File: lib/src/stack/coap_ientry.dart [hunk 3]
```diff
@@ -8,22 +8,22 @@
 /// Represents a name-filter pair that an <see cref="IChain&lt;TFilter, TNextFilter&gt;"/> contains.
 abstract class CoapIEntry<TFilter, TNextFilter> {
   /// Gets the name of the filter.
-  String? name;
+  late String name;
```
[the remark this answers is not in the document]
Comment (JosefWN) [reply lib/src/stack/coap_ientry.dart]: Fixed in: https://github.com/shamblett/coap/pull/95/commits/ec979c4406db8839059f009f8327cae884b5788e

**Review by JKRhb — APPROVED** [review]
Thanks for addressing the comments, @JosefWN :) Can you also have a look, @shamblett?

**Comment by shamblett** [comment]
Looks OK to me, I can merge this if we are agreed, also we should be thinking of re releasing the package at this point, quite a bit ahs gone in since the last release.

**Comment by shamblett** [comment]
OK merged, all the analysis errors have now gone, thanks.

**Pull request merged**